llvm/lib/Target/RISCV/RISCVISelDAGToDAG.cpp

```diff
@@ -57,6 +57,28 @@ void RISCVDAGToDAGISel::PreprocessISelDAG() {
 
     SDValue Result;
     switch (N->getOpcode()) {
+    case ISD::BITCAST: {
```

Comment:
Do we still need the preprocessing for the bitcast? What happens if this bit is removed

Reply:
If this is removed, we need to process this in `select` function which need to expand `VMV_V_X_VL` manually which is more complicated

Reply:
There is a subregister hiearchy for tuples. But if the NF isn't a valid LMUL, I don't know if it would work.

Reply:
But I think we don't model the hierarchy like `VRN2M2 <-> VRM4` right?

Reply:
> But I think we don't model the hierarchy like VRN2M2 <-> VRM4 right?

`copyPhysReg` might work if it just looks up the start and number of registers for each of the register classes. But there's no equivalence when NF is not a power of 2.

Reply:
I think it can't, it needs both src and dst registers to be in same register class